pytket/qir/conversion/conversion.py

```diff
@@ -394,33 +413,36 @@ def _to_qis_bits(self, args: List[Bit]) -> Sequence[Value]:
 
     def _reg2ssa_var(self, bit_reg: BitRegister, int_size: int) -> Value:
         """Convert a BitRegister to an SSA variable using pyqir types."""
-        reg_name = bit_reg[0].reg_name
-        if (
-            reg_name not in self.ssa_vars.keys()
-        ):  # Check if the register has been previously set.
+        reg_name = bit_reg.name
+        if reg_name not in self.ssa_vars.keys():
             # Check if the register has been previously set. If not, initialise to 0.
+            empty_reg = False
```

Comment:
Why is it important to handle this as a special case?
What is stopping us from just not converting an empty register?

Reply:
I did not understand this question fully. In the existing code there is a check for whether the classical register has been written into (set), so I am only introducing `init_reg` for when that's not the case, ie, for an empty register.

We could completely remove `reg2var` and use `init_reg` (or something with an even better name) to pass in a value other than zero. This is something we can discuss.